tests/python/pants_test/backend/python/tasks/test_pytest_run.py

```diff
@@ -611,12 +606,12 @@ def test_coverage_auto_option_red(self):
     self.assertEqual([1, 2, 5, 6], all_statements)
     self.assertEqual([2], not_run_statements)
 
-  @ensure_cached(PytestRun, expected_num_artifacts=0)
+  @ensure_cached(PytestRun, expected_num_artifacts=1)
   def test_coverage_auto_option_mixed_multiple_targets(self):
     all_statements, not_run_statements = self.run_coverage_auto(targets=[self.green, self.red],
                                                                 failed_targets=[self.red])
     self.assertEqual([1, 2, 5, 6], all_statements)
-    self.assertEqual([], not_run_statements)
+    self.assertEqual([2], not_run_statements)
```

Comment:
I couldn't figure this one out. 

This is now the same output as `test_coverage_auto_option_red`. I suspect what's happening is that `self.green` says every statement ran, but `self.red` says statement 2 did not, so the combined result says statement 2 did not run?

cc @jsirois, do you recall if V1 Pytest `--no-fast` will attempt to merge coverage results? I know that's a tricky task @TansyArron has spent much time on for V2.